packages/core/src/contrib/Constraints.ts

```diff
@@ -229,6 +229,54 @@ export const objDict = {
   nearScalar: (c: any, goal: any) => {
     return squared(sub(constOfIf(c), constOfIf(goal)));
   },
+  /**
+   * Repel the angle between the p1-p0 and p1-p2 away from 0 and 180 degrees. NOTE: angles between 10 and 170 degrees are considered satisfied.
+   */
+  nonDegenerateAngle: (
+    [, p0]: [string, any],
+    [, p1]: [string, any],
+    [, p2]: [string, any],
+    strength = 20
+  ) => {
+    if (every([p0, p1, p2].map((props) => props["center"]))) {
+      const c0 = fns.center(p0);
+      const c1 = fns.center(p1);
+      const c2 = fns.center(p2);
+
+      const l1 = ops.vsub(c0, c1);
+      const l2 = ops.vsub(c2, c1);
+
+      const cosine = absVal(ops.vdot(ops.vnormalize(l1), ops.vnormalize(l2)));
+      // angles between 10 and 170 degrees do not need to be pushed
+      return ifCond(
+        lt(cosine, varOf(0.985)), // cos10 deg = .9848, abs(cos170) = .9848
```

Comment:
It would be good if we can take the min and max angles as arguments, perhaps with defaults `10` and `170`?

Reply:
I added a `range` variable that defines how many degrees from 0 or 180 are considered unsafe (it's 10 by default, so angles <10 degrees or >170 degrees are in the "unsafe" range)